xfn-gotemplate/main.go

```diff
@@ -0,0 +1,235 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/Masterminds/sprig/v3"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/crossplane/crossplane-runtime/pkg/fieldpath"
+	fnv1alpha1 "github.com/crossplane/crossplane/apis/apiextensions/fn/io/v1alpha1"
```

Comment:
I wonder if we should break this out of c/c (or replicate it out) into some Go SDK for building functions, to avoid everyone having to take a dependency on c/c.

Reply:
Can't we simply move it to `crossplane-runtime` and introduce a package (or packages) there as helpers for building functions?